File: 2021/solutionsR/day1.R
# Day 1 - Part 1 ----------------------------------------------------------

input <- as.integer(readLines("2021/data/day1.txt", warn = F)) 

length(which(diff(input) > 0)) 
# the length of vector of differences which are larger than 0

# an even shorter alternative:
sum(diff(input) > 0)

# Day 1 - Part 2 ----------------------------------------------------------

input <- as.integer(readLines("2021/data/day1.txt", warn = F))

length(which(diff(input, lag = 3) > 0))
# the length of vector of differences lagged by 3 which are larger than 0

# an even shorter alternative:
sum(diff(input, lag = 3) > 0)

File: 2019/solutionsPython/day1.py
# Imports -----------------------------------------------------------------
import numpy as np

# Day 1 - Part 1 ----------------------------------------------------------
with open("2019/data/day1.txt") as file:
    inp = [int(line.strip("\n")) for line in file]

inp = np.array(inp)
inp = np.floor(inp / 3) - 2

int(np.sum(inp))

# Day 1 - Part 2 ----------------------------------------------------------
with open("2019/data/day1.txt") as file:
    inp = [int(line.strip("\n")) for line in file]

inp = np.array(inp)
inp = np.floor(inp / 3) - 2

int(np.sum(inp))

